ui/tests/integration/components/terminal-test.ts

```diff
@@ -0,0 +1,28 @@
+import { module, test } from 'qunit';
+
+import { createTerminal } from 'waypoint/utils/terminal';
+import hbs from 'htmlbars-inline-precompile';
+import { later } from '@ember/runloop';
+import { render } from '@ember/test-helpers';
+import { setupRenderingTest } from 'ember-qunit';
+
+module('Integration | Component | terminal', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('basic rendering', async function (assert) {
+    // Setup terminal externally, as expected by the component
+    let terminal = createTerminal({ inputDisabled: true });
+    this.set('terminal', terminal);
+    // pass terminal and render
+    await render(hbs`<Terminal @terminal={{this.terminal}}/>`);
+    // write line and see if it renders
+    terminal.writeln('Welcome to Waypoint!');
+    // We have to use the runloop as writeln isn't async
+    // Note that even the xterm.js rendering tests use polling to evaluate rendering
+    later(() => {
+      assert.equal(terminal?.buffer?.active?.getLine(0)?.translateToString(true), 'Welcome to Waypoint!');
+    }, 300);
+    // Clean up terminal
+    terminal.dispose();
```

Comment:
Rather than calling `dispose` in the test, I think we could call `clearRender` and assert that dispose was called correctly by the component.

```suggestion
    // Clean up terminal
    await clearRender();
    assert.equal(terminal.element, undefined);
```

Reply:
The element is retained on the object on `dispose`, so now I'm using [`assignedSlot`](https://developer.mozilla.org/en-US/docs/Web/API/Element/assignedSlot) to assess whether it's been removed from the DOM.